vim/core/oni-markdown-preview/src/index.tsx

```diff
@@ -0,0 +1,150 @@
+import * as marked from "marked"
+import * as React from "react"
+
+/**
+ * Props are like the constructor arguments
+ * for the React component (immutable)
+ */
+export interface IMarkdownPreviewProps {
+    oni: Oni.Plugin.Api
+}
+
+export interface IMarkdownPreviewState {
+    source: string
+}
+
+export class MarkdownPreview extends React.PureComponent<IMarkdownPreviewProps, IMarkdownPreviewState> {
+    private _fs = require('fs')
+    private _subscriptions: Array<Oni.IDisposable> = []
+
+    constructor(props: IMarkdownPreviewProps) {
+        super(props)
+
+        this.state = { source: "" }
+    }
+
+    componentDidMount() {
+        const editor = this.props.oni.editors.activeEditor
+        this.subscribe(editor.onBufferChanged, (args) => this.onBufferChanged(args))
+        //TODO: Subscribe "onFocusChanged"
+
+        this.previewBuffer(this.props.oni.editors.activeEditor.activeBuffer)
+    }
+
+    componentWillUnmount() {
+        for (let subscription of this._subscriptions) {
+            subscription.dispose()
+        }
+        this._subscriptions = []
+    }
+
+    private subscribe<T>(editorEvent: Oni.IEvent<T>, eventCallback: Oni.EventCallback<T>): void {
+        this._subscriptions.push(editorEvent.subscribe(eventCallback))
+    }
+
+    private onBufferChanged(bufferInfo: Oni.EditorBufferChangedEventArgs): void {
+        if (bufferInfo.buffer.language == "markdown") {
+            this.previewBuffer(bufferInfo.buffer)
+        }
+    }
+
+    private previewBuffer(buffer: Oni.Buffer): void {
+        this.previewString("# Loading preview...")
+        buffer.getLines().then((lines: string[]) => {this.previewString(lines.join("\n"))})
+    }
+
+    private previewString(str: string): void {
+        this.setState({source: str})
+    }
+
+    public render(): JSX.Element {
+        const containerStyle: React.CSSProperties = {
+            padding: "1em 1em 1em 1em",
+        }
+
+        const html = marked(this.state.source)
+        return <div style={containerStyle} dangerouslySetInnerHTML={{__html: html}}></div>
```

Comment:
There might be a couple classes you want to add here: `className="stack enable-mouse"`. The first one causes it to be sized to fit the editor window (it sets `position: absolute` plus clamps the bounds to the parent), and `enable-mouse` will allow it to pick up mouse events.